### PR TITLE
Added timeout for TCP conns

### PR DESCRIPTION
**Pull request by leonsodhi**

### CODE REVIEW

File: syslog/syslog.go [hunk 1]
```diff
@@ -60,9 +60,12 @@ func dial(network, raddr string, rootCAs *x509.CertPool) (*conn, error) {
...
 			config = &tls.Config{RootCAs: rootCAs}
 		}
-		netConn, err = tls.Dial("tcp", raddr, config)
+		dialer := &net.Dialer{
+			Timeout : time.Duration(30) * time.Second,
+		}
+		netConn, err = tls.DialWithDialer(dialer, "tcp", raddr, config)
 	case "udp", "tcp":
-		netConn, err = net.Dial(network, raddr)
+		netConn, err = net.DialTimeout(network, raddr, time.Duration(30) * time.Second)
```
Comment (eric) [inline syslog/syslog.go]: We only need this for the tcp connection, not udp.

**Comment by eric** [comment]
I think we should also add `connectTimeout` to the `Logger` struct to allow us to configure this as well.

File: syslog/syslog.go (hunk 1 above)
Comment (leonsodhi) [inline syslog/syslog.go]: I didn't know this when I wrote the code, but it sounds like we should keep this in even for UDP as it [also covers name resolution](https://github.com/golang/go/blob/f78a4c84ac8ed44aaf331989aa32e40081fd8f13/src/net/dial.go#L190).

Comment (eric) [inline syslog/syslog.go]: Works for me.

**Comment by leonsodhi** [comment]
Sorry, I made this change and then saw your comment. I'm not sure if #98 existed when you wrote this, so do you still want separate config options for a connection timeout and a write timeout?

**Comment by eric** [comment]
I think we should use a separate config option for a connect timeout and a write timeout.

File: .travis.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 language: go
 go:
-  - 1.2
+  - 1.3
```
Comment (troy) [inline .travis.yml]: :+1:

**Comment by troy** [comment]
If 2 timeouts are used, those names - `connect_timeout` and `write_timeout` - were clear to me.

**Comment by troy** [comment]
Looks good to me.

**Comment by troy** [comment]
Looks good to me

**Comment by eric** [comment]
:+1: Looks good to me.

**Pull request merged**